### ISSUE TITLE
`terraform plan -destroy` lists data resources

### ISSUE THREAD

**Issue by ThisGuyCodes** [hashicorp/terraform#14048]
As I understand it, `data.*` objects don't get deleted, or even really controlled, by terraform; they're used to pull data from existing resources.

But `terraform plan -destroy` lists data objects as "to be deleted". While I don't believe terraform ever manipulates them this is incredibly unnerving to see, and also incorrect.

### Terraform Version
v0.9.3

### Affected Resource(s)
Please list the resources as a list, for example:
- data.*

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-east-1"
}

data "aws_ami" "ubuntu" {
  most_recent = true

  owners = ["099720109477"] # Canonical

  filter {
    name   = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-yakkety-16.10-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }
}
```

### Expected Behavior
Data objects don't get deleted (I hope?) so they shouldn't be listed.

### Actual Behavior
Data objects are listed as if they're going to be deleted.

### Steps to Reproduce
1. `terraform plan -destroy`

**Comment by muffadalis** [comment]
I had similar experience. Being a noob, I got terrified on see that _destroy_ will destroy by VPC & Subnets and not just the EC2. To be fair, terraform plan did say "data.*", may be sort more documentation would be nice.

**Comment by apparentlymart** [comment]
Hi @ThisGuyCodes and @muffadalis! Sorry for the confusion here.

This is indeed a bug. Internally some values are deleted from Terraform's state here, which is why they are included in the plan. But you're totally right that it doesn't make any sense for this detail to be visible in the UI.

I expect this was a missed case in the 0.9 graph-builder rewriting, since before 0.9 deletion of data resources from state happened in a different way, which caused some bugs (causing this to get rearranged in 0.9) but kept the items out of the user-facing plan.

**Comment by synhershko** [comment]
this is still the case also with v0.10.0-rc1-dev - when calling `terraform destroy` data items are also being listed, and it isn't graceful

**Comment by apparentlymart** [comment]
From looking at the family of issues here, it looks like there are two different things to fix that might as well get fixed together:

* The plan includes `-` actions for data resources.
* The apply output mentions that it's running the destroy action on data resources.

The only message we should ever see about a data source is that it's being refreshed. That's either the `<=` action in the plan, or the `Refreshing state...` message in the action log.

**Comment by miketruty** [comment]
Thanks for the redirect.  Let me say this differently, from a customer perspective.

If I run plan -destroy and it says "I'm going to destroy 32 resources", then I run destroy and it says "I just destroyed 34 resources" it's really scary and unnerving.

Holy cow, I just deleted 2 extra resource?!?  Which ones?!  Why didn't it tell me?!

Destroy complete! Resources: 34 destroyed.

Is there any intent to align the two counts?  That would be my ask.

**Comment by apparentlymart** [comment]
Thanks for that, @miketruty... indeed the counts ought to line up too. The counts are driven by the same event that produces the log lines during apply, so fixing one should fix the other. Good to be explicit about that though: we shouldn't consider this fixed until both the counts _and_ the log output are updated as described above.

**Comment by dev-head** [comment]
sanity work around: delete your terraform config, run plan/apply, replace deleted config with proper config and start over. 

I'm over here like, i am pretty darn sure we're good to delete these data resources because it's not controlled by Terraform. but i'll be damned if we drop production on a hunch.

**Comment by apparentlymart** [comment]
Hi all! These issues are addressed by #15884, which should be included in the next release. Sorry again for all the confusing output here, and thanks for your patience while we got this fixed.
